pam/return.go

```diff
@@ -1,54 +1,41 @@
 package main
 
+import (
+	"github.com/msteinert/pam"
+)
+
 // Various signalling return messaging to PAM.
 
-// pamSuccess signals PAM module to return PAM_SUCCESS and Quit tea.Model.
+type pamReturnStatus interface {
+	Status() pam.StatusError
+	Error() string
+}
+
+// TODO: Remove this when we've tests coverage that ensures this is the case.
+// ensure that [pam.NewTransactionError] implements [pamReturnStatus].
+var _ pamReturnStatus = pam.NewTransactionError(pam.ErrAbort, nil)
+
+// pamStatus signals PAM module to return with provided pam.Success and Quit tea.Model.
 type pamSuccess struct {
+	pam.TransactionError
 	brokerID string
 }
 
-// String returns the string of pamSuccess.
-func (err pamSuccess) String() string {
-	return ""
+// newPamSuccess returns a new pamSuccess.
+func newPamSuccess(brokerID string) pamReturnStatus {
```

Comment:
never return an interface from any function, accept interface, return concrete types.

Reply:
Oh, I see... Well I thought it was a good way to ensure that the type actually implements the interface hiding its actual type.

Also, isn't this what we do when we return a specific error type while the function returns just an `error`?

Reply:
`error` is really the exception for the rule (the only interface we return), but it’s part of stdlib. Then, we don't type assert explicitely apart from `errors.Is()`.